demos/object_detection_demo/cpp/main.cpp

```diff
@@ -117,6 +119,78 @@ static void showUsage() {
     std::cout << "    -yolo_af                  " << yolo_af_message << std::endl;
 }
 
+class ColorPalette {
+private:
+    std::vector<cv::Scalar> palette;
+
+    double getRandom(double a = 0.0, double b = 1.0) {
+        static std::default_random_engine e;
+        std::uniform_real_distribution<> dis(a, std::nextafter(b, std::numeric_limits<double>::max()));
+        return dis(e);
+    }
+
+    double distance(const cv::Scalar& c1, const cv::Scalar& c2) {
+        auto dh = std::fmin(std::fabs(c1[0] - c2[0]), 1 - fabs(c1[0] - c2[0])) * 2;
+        auto ds = std::fabs(c1[1] - c2[1]);
+        auto dv = std::fabs(c1[2] - c2[2]);
+
+        return dh * dh + ds * ds + dv * dv;
+    }
+
+    cv::Scalar maxMinDistance(const std::vector<cv::Scalar>& colorSet, const std::vector<cv::Scalar>& colorCandidates) {
+        std::pair<size_t, double> maxDist{ 0, 0 };
+        size_t i = 0;
+        for (auto& c1 : colorCandidates) {
+            double minDist = std::numeric_limits<double>::max();
+            for (auto& c2 : colorSet) {
+                auto dist = distance(c1, c2);
+                minDist = dist < minDist ? dist : minDist;
+            }
+            maxDist = minDist > maxDist.second ? std::pair<size_t, double>{i, minDist} : maxDist;
+            ++i;
+        }
+
+        return colorCandidates[maxDist.first];
+    }
+
+    cv::Scalar hsv2rgb(cv::Scalar& hsvColor) {
```

Comment:
```suggestion
    static cv::Scalar hsv2rgb(const cv::Scalar& hsvColor) {
```

(the argument should be const, as it's illogical for a function like this to modify its input argument)

Reply:
Done.